File: ag-ui/integrations/claude-agent-sdk/typescript/examples/agentic_chat.ts
/**
 * Agentic chat example - basic configuration.
 *
 * This example shows how to create a basic agentic chat adapter
 * using the Claude Agent SDK integration.
 */

import { ClaudeAgentAdapter } from "@ag-ui/claude-agent-sdk";
import { DEFAULT_DISALLOWED_TOOLS } from "./constants";

/**
 * Create adapter for agentic chat.
 *
 * The adapter configuration supports all Claude SDK Options.
 * See: https://platform.claude.com/docs/en/agent-sdk/typescript
 */
export function createAgenticChatAdapter(): ClaudeAgentAdapter {
  return new ClaudeAgentAdapter({
    agentId: "agentic_chat",
    description: "General purpose agentic chat assistant",
    model: "claude-haiku-4-5",
    systemPrompt: "You are a helpful assistant with access to tools.",
    includePartialMessages: true,
    disallowedTools: [...DEFAULT_DISALLOWED_TOOLS],
  });
}

File: ag-ui/integrations/claude-agent-sdk/python/examples/agents/agentic_chat.py
"""
Agentic chat agent configuration.

This module provides a factory function for creating an agentic chat adapter.
The adapter supports all ClaudeAgentOptions from the Claude Agent SDK.
"""

from ag_ui_claude_sdk import ClaudeAgentAdapter
from .constants import DEFAULT_DISALLOWED_TOOLS


def create_agentic_chat_adapter() -> ClaudeAgentAdapter:
    """Create adapter for agentic chat."""
    return ClaudeAgentAdapter(
        name="agentic_chat",
        description="General purpose agentic chat assistant",
        options={
            "model": "claude-haiku-4-5",
            "system_prompt": "You are a helpful assistant with access to tools.",
            "disallowed_tools": list(DEFAULT_DISALLOWED_TOOLS),
        }
    )
